.github/workflows/java24.yml

```diff
@@ -0,0 +1,30 @@
+name: Integration Tests on Java 24
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 24
+        uses: actions/setup-java@v3
+        with:
+          java-version: 24
+          distribution: 'temurin'
+      - name: Switch To Maven 3.9.10
+        run: |
+          echo "distributionUrl=https://dlcdn.apache.org/maven/maven-3/3.9.10/binaries/apache-maven-3.9.10-bin.zip" > .mvn/wrapper/maven-wrapper.properties
+          echo "wrapperUrl=https://repo.maven.apache.org/maven2/org/apache/maven/wrapper/maven-wrapper/3.3.2/maven-wrapper-3.3.2.jar" >> .mvn/wrapper/maven-wrapper.properties
+      - name: Build with Maven
+        run: ./mvnw --batch-mode --update-snapshots install -Prun-its
+      - uses: actions/upload-artifact@v3
```

Comment:
Looks like this should now use v4 as v3 is deprecated.

https://github.blog/news-insights/product-news/get-started-with-v4-of-github-actions-artifacts/

I'll update the other workflows, as it also seems 3.9.5 can't be downloaded now.

I'll likely manually squash/merge these commits into some other pending commits I have as well.

Reply:
I've merged changes to the dependencies and workflows, so you should be able to rebase this PR on top of `master`.

We should probably split that new workflow into its own PR as well.  I'll do that now actually.

Reply:
The updated workflow has been merged - not sure why they're not automatically running on your PR tho.  Rebase away.